src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -43,18 +44,17 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                         nameof(actionContext));
             }
 
+
+            var methodParameters = actionDescriptor.MethodInfo.GetParameters();
             var parameterMetadata = new List<ModelMetadata>();
             foreach (var parameter in actionDescriptor.Parameters)
             {
+                var parameterInfo = methodParameters.Where(p => p.Name == parameter.Name).Single();
                 var metadata = _modelMetadataProvider.GetMetadataForParameter(
-                    methodInfo: actionDescriptor.MethodInfo,
```

Comment:
What was the reason for replacing method info with parameterinfo? We could have followed a model similar to what we do for properties.